File: targets/numworks/tests/camlboot/benchs/integrales_tr.py
import time

def integrale(f, a, b, n):
    accu = 0.0
    x = a
    h = (b - a) / n
    while (x< b):
        accu = accu + f(x) ;
        x = x+ h
    return accu
  
def poly(x):
    return x * x + 2 * x + 1

def test_integrales(n):
    r = 0.0
    i = 0
    while (i< n):
        r= r + integrale(poly,0.0,1.0,100)
        i = i + 1
    return r

start = time.monotonic()
test_integrales(4)
stop = time.monotonic()
print(stop-start)

File: targets/numworks/tests/camlboot/benchs/integrales_tr.ml
let rec integ accu x f  b h =
  if  x >=  b then accu else integ (accu +. (f x)) (x +. h) f  b h
;;

let integrale f a b n =
  let h = (b -. a) /. (float_of_int n) in
  integ  0.0 (a *. h) f  b h 
;;


let poly x = x *. x +. 2. *. x +. 1.
;;

let repeat n =
  for i = 0 to n do
    let _ = integrale poly 0.0 1.0 100 in () (* XXX I changed n, otherwise out of memory *)
  done
;;

(* integrale poly 0.0 1.0 10000 -> 2.333583335 (* 0.333383335 *) *)

let () =
  let start = millis () in
  repeat 4; (* x 1000 for OMicroB *)
  let stop = millis () in
  print_int (stop-start)
